File: src/main/java/de/simpletactics/wiki/lib/service/port/WikiRightsPort.java
package de.simpletactics.wiki.lib.service.port;

public interface WikiRightsPort {


  String getWikiBerechtigung(String username, int wikiId);

  void deleteWikiDefaultRead(int wikiId);

  void deleteWikiAllRights(int wikiId);

  void setWikiRightsCopyOvertopic(int wikiId, int overtopicId);

  int getWikiDefaultRead(int wikiId);

  void addWikiUserRight(int wikiId, String username, String wikiRight, boolean recursive);

  void deleteWikiUserRight(int wikiId, String username);

  void addWikiRoleRight(int wikiId, int role, String wikiRight, boolean recursive);

  void setWikiDefaultRead(int wikiId, int roleId);
}

File: src/main/kotlin/de/simpletactics/wiki/lib/services/port/WikiRightsPort.kt
package de.simpletactics.wiki.lib.services.port

import de.simpletactics.wiki.lib.model.Right

interface WikiRightsPort {

    fun getWikiBerechtigung(username: String, wikiId: Int): Right?

    fun deleteWikiDefaultRead(wikiId: Int)

    fun deleteWikiAllRights(wikiId: Int)

    fun setWikiRightsCopyOvertopic(wikiId: Int, overtopicId: Int)

    fun getWikiDefaultRead(wikiId: Int): Int

    fun addWikiUserRight(wikiId: Int, username: String, wikiRight: String, recursive: Boolean)

    fun deleteWikiUserRight(wikiId: Int, username: String)

    fun addWikiRoleRight(wikiId: Int, role: Int, wikiRight: String, recursive: Boolean)

    fun setWikiDefaultRead(wikiId: Int, roleId: Int)

}